cares-center-inventory-Song/crud.py

```diff
@@ -0,0 +1,242 @@
+
+def delete_manager_by_manager_id(db: Session, manager_id: str):
+    manager_to_delete = db.query(models.Manager).filter(models.Manager.manager_id == manager_id).first()
+    db.delete(manager_to_delete)
+    db.commit()
+
+def verify_manager(db: Session, username: str, password: str):
+    manager_found =  db.query(models.Manager).filter(and_(or_(models.Manager.manager_id == username, models.Manager.manager_name == username), models.Manager.passcode == password)).first()
+    return False if not manager_found else True
+
+# donors
+def create_donor(db: Session, donor: schemas.DonorCreate):
+    db_donor = models.Donor(
+        donor_name = donor.donor_name
+    )
+    db.add(db_donor)
+    db.commit()
+    db.refresh(db_donor)
+    return db_donor
+
+def get_donor_by_donor_name(db: Session, donor_name: str):
+    return db.query(models.Donor).filter(models.Donor.donor_name == donor_name).first()
+
+def get_donors(db: Session):
+    return db.query(models.Donor).all()
+
+def update_donor_name_by_donor_id(db: Session, donor_id: str, new_donor_name: str):
+    db.query(models.Donor).filter(models.Donor.donor_id == donor_id).update({models.Donor.donor_name: new_donor_name})
+    db.commit()
+
+def delete_donor_by_donor_name(db: Session, donor_name: str):
+    donor_to_delete = db.query(models.Donor).filter(models.Donor.donor_name == donor_name).first()
+    db.delete(donor_to_delete)
+    db.commit()
+
+# items
+def create_item(db: Session, item: schemas.ItemCreate):
+    item_barcodes = [item.item_barcode for item in db.query(models.Item).all()]
+    while True:
+        new_barcode = str(randint(0, 9999999)).rjust(7, '0')
+        if new_barcode not in item_barcodes:
+            break
+
+    db_item = models.Item(
+        item_name = item.item_name,
+        item_points = item.item_points,
+        item_front_quantity = item.item_front_quantity,
+        #item_back_quantity = item.item_back_quantity,
+        donor_id = item.donor_id,
+        item_barcode = new_barcode
+    )
+    db.add(db_item)
+    db.commit()
+    db.refresh(db_item)
+    return db_item
+
+def get_items(db: Session):
+    return db.query(models.Item).all()
+    
+def get_items_by_donor_id(db: Session, donor_id: str):
+    return db.query(models.Item).filter(models.Item.donor_id == donor_id).all()
+
+def get_items_by_donor_name(db: Session, donor_name: str):
+    return db.query(models.Item).join(models.Donor).filter(models.Donor.donor_name == donor_name).all()
+
+def get_items_by_point_interval(db: Session, lhs: int, rhs: int):
+    return db.query(models.Item).filter(and_(models.Item.item_points >= lhs, models.Item.item_points <= rhs)).all()
+
+def update_item_name_by_item_id(db: Session, item_id: str, new_item_name: str):
+    db.query(models.Item).filter(models.Item.item_id == item_id).update({models.Item.item_name: new_item_name})
+    db.commit()
+
+def update_item_front_quantity_by_item_id(db: Session, item_id: str, new_item_front_quantity):
```

Comment:
duplicate this for back quantity, item_id should be int, new_item_front_quantity should be float